store/postgres/src/entities.rs

```diff
@@ -792,14 +841,50 @@ impl Connection {
         self.conn.batch_execute(&*query)?;
 
         match *GRAPH_STORAGE_SCHEME {
-            v::Relational => Layout::create_relational_schema(
-                &self.conn,
-                &schema_name,
-                schema.id.clone(),
-                &schema.document,
-            )
-            .map(|_| ()),
-            v::Split => create_split_schema(&self.conn, &schema_name),
+            v::Relational => {
+                let layout = Layout::create_relational_schema(
+                    &self.conn,
+                    &schema_name,
+                    schema.id.clone(),
+                    &schema.document,
+                )?;
+                // See if we are grafting and check that the graft is permissible
+                if let Some((base, _)) = metadata::deployment_graft(&self.conn, &schema.id)? {
+                    match Storage::new(&self.conn, &base)? {
+                        Storage::Relational(base) => {
+                            let errors = layout.can_copy_from(&base);
```

Comment:
I'd be loathe to do it with the tools we have for handling GraphQL schemas right now. A big difference between a `Layout` and the GraphQL `Document` is that a lot of potential problems are clearly not possible anymore with a `Layout`: for example, when you have a `Layout`, you know that each attribute has a valid type. When you deal with a GraphQL `Document` every single time you need to look at the type of an attribute, you have to be prepared for the fact that the user might have typoed `Account` as `Accuont` Even if the code flow says that it was checked before, is it really ok to use `unwrap` or `expect` in all those cases? It embeds a lot of knowledge along the lines of 'the data structure makes this possible, but we know from how the code flows that it's really not a problem here' and results in GraphQL schema validation code that is very hard to follow.

Reply:
Yeah, I know that the raw GraphQL schema is awkward to work with. At the same time, decisions like whether two schemas are compatible should not be made deep down in the store either. The fact that even Graph CLI could check for schema compatibility also speaks against tying it so closely to the store implementation.

It could be nice if we translated the unvalidated schema `Document` into a validated `Schema` type that has an unambiguous, checked and convenient shape. The translation from that to a `Layout` should be straight-forward, with no potential for errors.

But... that's for another day.